include/ddsrouter/types/topic/WildcardTopic.hpp

```diff
@@ -36,7 +36,8 @@ class WildcardTopic : public FilterTopic
 
     //! Constructor that allows any type
     WildcardTopic(
-            const std::string& topic_name);
+            const std::string& topic_name,
+            bool topic_with_key = false);
```

Comment:
Do not set this here and add an attribute so FilterTopics has `keyed` or `not_keyed` and also store `has_keyed_set`.
In case the keyed is not been set, the filter should apply to both.

Reply:
:+1: